### DISCUSSION TITLE
NTSerializer only encodes in ASCII

### DISCUSSION THREAD

**Discussion by tcmitchell** [RDFLib/rdflib#1615] [Q&A] [answered]
The NTSerializer only encodes in ASCII and issues a warning if the encoding is anything other than `"ascii"`. See [nt.py line 22](https://github.com/RDFLib/rdflib/blob/master/rdflib/plugins/serializers/nt.py#L22) and [nt.py lines 27-28](https://github.com/RDFLib/rdflib/blob/master/rdflib/plugins/serializers/nt.py#L27-L28). See also #1376 

The [N-Triples Grammar](https://www.w3.org/TR/n-triples/#n-triples-grammar) says:

> An N-Triples document is a Unicode [UNICODE] character string encoded in UTF-8.

This may be a change from earlier (pre-2014) versions of N-Triples. See the first bullet in [Section 3 Changes from RDF Test Cases format](https://www.w3.org/TR/n-triples/#n-triples-changes). But this change is 7 years old.

Example 3 in [Section 2.3 RDF Literals](https://www.w3.org/TR/n-triples/#sec-literals) shows unicode characters in an example N-Triples document.

There is a mention of ASCII encoding in [Section 6.1 Other Media Types](https://www.w3.org/TR/n-triples/#sec-other-media-types) but that is for passing N-Triples documents as `text/plain`. Section 6.1 goes on to say that "In ... these cases the document is _not_ an N-Triples document" (emphasis mine).

**Comment by white-gecko** [comment]
This shall be solved with #1222

**Comment by tcmitchell** [comment, marked as answer]
Aha! More source diving revealed `NT11Serializer`, which is not limited to ASCII. Why isn't this the default N-Triples serializer? Why stick with the ASCII-only, and ancient, `NTSerializer`?

See https://github.com/RDFLib/rdflib/issues/1376#issuecomment-885741449
